test/Language/Github/Actions/WorkflowTest.hs

```diff
@@ -50,13 +49,13 @@ test_goldenWorkflowFromYaml = do
               putStrLn $ "roundtrip " <> takeBaseName testYamlFilePath
               eitherWorkflow <- Yaml.decodeFileEither @Workflow testYamlFilePath
               either
-                (BS.writeFile outputFilePath >> (\e -> fail $ "YAML decoding failed: " ++ show e))
+                (writeFile outputFilePath >> (\e -> fail $ "YAML decoding failed: " ++ show e))
                 (\workflow -> writeOutputFiles outputFilePath haskellOutputFilePath workflow >> pure workflow)
-                $ first (encodeUtf8 . Text.pack . Yaml.prettyPrintParseException) eitherWorkflow
+                $ first Yaml.prettyPrintParseException eitherWorkflow
     writeOutputFiles :: FilePath -> FilePath -> Workflow -> IO ()
     writeOutputFiles outputFilePath haskellOutputFilePath workflow = do
-      BS.writeFile outputFilePath (Yaml.encode workflow)
-        >> BS.writeFile haskellOutputFilePath (encodeUtf8 . Text.pack $ ppShow workflow)
+      writeFile outputFilePath (Text.unpack . decodeUtf8 $ Yaml.encode workflow)
```

Comment:
This is wasteful: it turns a lazy sequence of bytestring chunks (good!) into text, then into a singly-linked list of characters, just to send it out to disk.

I would revert this change.